livekit-plugins/livekit-plugins-avatar/livekit/plugins/avatar/io.py

```diff
@@ -0,0 +1,223 @@
+        """Capture and stream audio frame to remote worker"""
+        if not self._remote_participant:
+            raise RuntimeError("Worker participant not found")
+
+        if not self._stream_writer:
+            # start new stream
+            # TODO: any better option to send the metadata?
+            name = f"audio_{frame.sample_rate}_{frame.num_channels}"
+            self._stream_writer = await self._room.local_participant.stream_file(
+                file_name=name,
+                mime_type="audio/raw",
+                destination_identities=[self._remote_participant.identity],
+            )
+        await self._stream_writer.write(frame.data)
+
+        await super().capture_frame(frame)
+
+    async def flush(self) -> None:
+        """Mark end of current audio segment"""
+        super().flush()  # TODO: should the flush and clear_buffer async?
+
+        if self._stream_writer is None:
+            return
+
+        # close the stream marking the end of the segment
+        await self._stream_writer.aclose()
+        self._stream_writer = None
+
+    async def clear_buffer(self) -> None:
+        """Stop current stream immediately"""
+        assert self._remote_participant is not None
+
+        self.__nb_playback_finished_needed = 0  # Is this needed?
+        await self._room.local_participant.perform_rpc(
+            destination_identity=self._remote_participant.identity,
+            method=RPC_INTERRUPT_PLAYBACK,
+            payload="",
+        )
+
+
+class AudioFlushSentinel:
+    pass
+
+
+@dataclass
+class _AudioReader:
+    stream_reader: rtc.FileStreamReader
+    sample_rate: int
+    num_channels: int
+
+
+class AudioReceiver(rtc.EventEmitter[Literal["interrupt_playback"]]):
+    """
+    Audio receiver that receives streamed audio from a sender participant using LiveKit DataStream.
+    Used by the worker to receive audio frames from a sender identified by the DATASTREAM_AUDIO_SENDER attribute.
+    """
+
+    def __init__(self, room: rtc.Room):
+        self._room = room
+        self._remote_participant: Optional[rtc.RemoteParticipant] = None
+
+        self._stream_readers: list[_AudioReader] = []
+        self._stream_received: asyncio.Event = asyncio.Event()
+
+    async def start(self) -> None:
+        """
+        Wait for sender participant to join and start receiving.
+        Also marks this participant as a receiver.
+        Usage:
+            audio_receiver = AudioReceiver(room)
+            await audio_receiver.start()
+            async for frame in audio_receiver.stream():
+                if isinstance(frame, AudioFlushSentinel):
+                    # segment completed
+                    pass
+                else:
+                    # process frame
+                    pass
+        """
+        # mark self as receiver
+        await self._room.local_participant.set_attributes({AUDIO_RECEIVER_ATTR: "true"})
+
+        self._remote_participant = await wait_for_participant(
```

Comment:
it seems here we can just wait for participant.kind == agent?

if we wanted to handle multiple avatars in the room, then the integration should take in the controller's identity.